notebooks/ndd_example.ipynb

```diff
@@ -0,0 +1,202 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "ed9dd0a2-6b12-436f-b6c3-02e0072115e3",
+   "metadata": {},
+   "source": [
+    "This notebook shows the results of query-time near-duplicate detection (NDD).  It shows the same query with and without the duplicates.  To use this notebook, first start the Sycamore containers using `docker compose up`.  Then, ingest the college credit card marketing agreement database.  There are two choices here.  The quickest approach is to ingest only the 35 documents that are needed for the query below.  This can be accomplished by running `docker compose run sycamore_crawler_s3 -anon aryn-public cccmad-tiny`.  The other option is to ingest all 1911 PDFs that have text, by running `docker compose run sycamore_crawler_s3 -anon aryn-public cccmad`.  This will take hours and around 20 PDFs will have problems.  Once ingestion has sufficiently completed, run this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "bbf93d45-bee3-4202-8d5a-88497196db06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import requests\n",
+    "import warnings\n",
+    "import urllib3\n",
+    "warnings.filterwarnings(\"ignore\", category=urllib3.exceptions.InsecureRequestWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "06e9a476-134f-4901-a89e-3c91bdb8881c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_model_id():\n",
+    "    query = {\n",
+    "        'query': {\n",
+    "            'bool': {\n",
+    "                'must': [\n",
+    "                    {\n",
+    "                        'match': {'name': 'all-MiniLM-L6-v2'},\n",
+    "                    },\n",
+    "                    {\n",
+    "                        'term': {'model_config.model_type': 'bert'},\n",
+    "                    },\n",
+    "                ],\n",
+    "            },\n",
+    "        },\n",
+    "    }\n",
+    "    with requests.get('https://opensearch:9200/_plugins/_ml/models/_search', json=query, verify=False) as resp:\n",
+    "        res = json.loads(resp.text)\n",
+    "        return res['hits']['hits'][0]['_id']"
+   ]
+  },
```

Comment:
Is it worth putting a note here explaining why we are doing this? We could link to the section of the docs that discusses this from the note, too.